cpp/src/structs/utilities.cpp

```diff
@@ -337,6 +342,116 @@ void superimpose_parent_nulls(bitmask_type const* parent_null_mask,
   }
 }
 
+namespace {
+
+/**
+ * @brief Functor to fetch a column-view's `head()` pointer.
+ *
+ * Required because `column_view::head<T>()` is a function template
+ * that necessitates type dispatch.
```

Comment:
You can pass `void` as the type for `head` to avoid needing to type dispatch. In fact, the default is `void` so you can just call `col.head()` without a template param.

Reply:
Thanks, Jake. That makes sense. Let me try that.

Reply:
This works well. I should've guessed I was missing something.